### PR TITLE
handles setting a new object on existing leaf array correctly

### LINKED ISSUE #151: setIn creates extra array in certain situation, w strange behavior
setIn seems to create an array, if setting an object property on an array that already exists. Although it seems like it's an array, that's an object? Or something of the sort, I'm not sure, I've never seen this before... it's odd. Screenshots to show what I mean

```
var x = Immutable({foo: []})
x.setIn(['foo', 0, 'bar'], 'val')

// results in
{
    foo: [
        [ ... array object thing?
            bar: val
        ]
    ]
}
```

screen shot 2016-09-20 at 7 03 17 pm

When stringified, it's as if setIn never happened...

screen shot 2016-09-20 at 7 08 27 pm

Confirmed in Chrome 52.0.2743.116, OS X, latest seamless-immutable.

### PR DESCRIPTION
**Pull request by marbemac**
closes #151

I'm not terribly familiar with the testing suite, so please let me know if there's a better way to do this. For examples sake, I included a test that demonstrates the issue. This test fails on the latest published seamless-immutable, and is fixed w this commit.

### CODE REVIEW

File: src/seamless-immutable.js [hunk 1]
```diff
@@ -127,7 +127,13 @@
         // Might (validly) be object or array
         newValue = thisHead.setIn(tail, value);
       } else {
-        newValue = arraySetIn.call(immutableEmptyArray, tail, value);
+        var nextHead = tail[0];
+        // If the next path part is a number, then we are setting into an array, else an object.
+        if (!isNaN(parseFloat(nextHead)) && isFinite(nextHead)) {
```
Comment (rtfeldman) [inline src/seamless-immutable.js]: Don't we want something other than `parseFloat` here? For example: http://stackoverflow.com/questions/12227594/which-is-better-numberx-or-parsefloatx

Comment (marbemac) [inline src/seamless-immutable.js]: Sure, we could use Numberx - something like this?

```js
if (nextHead !== '' && !isNaN(Number(nextHead)) && isFinite(nextHead)) {
```

After reading a bit more about how isFinite works, we could probably actually shorten it to this:

```js
if (nextHead !== '' && isFinite(nextHead)) {
```

Comment (rtfeldman) [inline src/seamless-immutable.js]: yeah - if that works, it seems best!

**Comment by marbemac** [comment]
Alrighty, updated! I included the dist files as well, but just realized you merged some changes into master earlier today - how would you like to handle it?

**Comment by rtfeldman** [comment]
@marbemac don't worry about the dist files; I always rebuild them before publishing. 😄

**Comment by marbemac** [comment]
Sweet! Thanks for reviewing this so quickly.

**Comment by rtfeldman** [comment]
@marbemac can you reset the dist files to what's on `master` and push that to resolve the merge conflict?

**Comment by rtfeldman** [comment]
Thanks @marbemac! Published as 6.1.3.

**Pull request merged**